core/src/main/java/org/radargun/traits/Queryable.java

```diff
@@ -27,27 +27,52 @@
 
    /**
     * The instance should be reusable, but not thread-safe.
+    * Conditions defined after groupBy call are meant to be part of the HAVING clause.
     */
    interface QueryBuilder {
       QueryBuilder subquery();
-      QueryBuilder eq(String attribute, Object value);
-      QueryBuilder lt(String attribute, Object value);
-      QueryBuilder le(String attribute, Object value);
-      QueryBuilder gt(String attribute, Object value);
-      QueryBuilder ge(String attribute, Object value);
-      QueryBuilder between(String attribute, Object lowerBound, boolean lowerInclusive, Object upperBound, boolean upperInclusive);
-      QueryBuilder isNull(String attribute);
-      QueryBuilder like(String attribute, String pattern);
-      QueryBuilder contains(String attribute, Object value);
+      QueryBuilder eq(SelectExpression selectExpression, Object value);
+      QueryBuilder lt(SelectExpression selectExpression, Object value);
+      QueryBuilder le(SelectExpression selectExpression, Object value);
+      QueryBuilder gt(SelectExpression selectExpression, Object value);
+      QueryBuilder ge(SelectExpression selectExpression, Object value);
+      QueryBuilder between(SelectExpression selectExpression, Object lowerBound, boolean lowerInclusive, Object upperBound, boolean upperInclusive);
+      QueryBuilder isNull(SelectExpression selectExpression);
+      QueryBuilder like(SelectExpression selectExpression, String pattern);
+      QueryBuilder contains(SelectExpression selectExpression, Object value);
       QueryBuilder not(QueryBuilder subquery);
       QueryBuilder any(QueryBuilder... subqueries);
-      QueryBuilder orderBy(String attribute, SortOrder order);
-      QueryBuilder projection(String... attribute);
+      QueryBuilder orderBy(SelectExpression selectExpression, SortOrder order);
```

Comment:
Infinispan doesn't allow you to do that, on the other hand it doesn't really follow any specification so 'legal' doesn't mean much.
For example I was just able to run
.select(Expression.property("accountId"), Expression.avg("amount"))
.groupBy("accountId")
.orderBy(Expression.max("amount"))
 but the result was not correct (at least compared to what I'd expect), and I don't know if SQL/JPQL allows this.